decentraland/screens/helpers.js

```diff
@@ -1,36 +1,175 @@
-  const mana = new Mana(MANA_ADDRESS);
+export const getContracts = () => {
+  let contracts;
+
+  const provider = ProviderFactory.getProvider();
+
+  const { _network: network } = provider;
+  const networkName = !network ? "local" : network.name;
+
+  const {
+    MANAToken,
+    LANDProxy,
+    EstateRegistry,
+    Marketplace,
+    GnosisSafe: GnosisSafeInfo,
+  } = TasitContracts[networkName];
+  const { address: MANA_ADDRESS } = MANAToken;
+  const { address: LAND_ADDRESS } = LANDProxy;
+  const { address: ESTATE_ADDRESS } = EstateRegistry;
+  const { address: MARKETPLACE_ADDRESS } = Marketplace;
+  const { address: GNOSIS_SAFE_ADDRESS } = GnosisSafeInfo;
 
-  mana.setWallet(fromAccount);
-  const approvalAction = mana.approve(toAddress, value.toString());
+  const { ERC20, ERC721, Marketplace: MarketplaceContracts } = Action;
+  const { Mana } = ERC20;
+  const { Estate, Land } = ERC721;
+  const { Decentraland: DecentralandMarketplace } = MarketplaceContracts;
+  const { GnosisSafe } = ContractBasedAccount;
 
-  await approvalAction.waitForNonceToUpdate();
+  const estateContract = new Estate(ESTATE_ADDRESS);
+  const landContract = new Land(LAND_ADDRESS);
+  const marketplaceContract = new DecentralandMarketplace(MARKETPLACE_ADDRESS);
+  const manaContract = new Mana(MANA_ADDRESS);
+  const gnosisSafeContract = new GnosisSafe(GNOSIS_SAFE_ADDRESS);
+
+  contracts = {
+    estateContract,
+    landContract,
+    marketplaceContract,
+    manaContract,
+    gnosisSafeContract,
+  };
+
+  return contracts;
 };
 
-export const manaFaucetTo = async (beneficiaryWallet, amountInWei) => {
-  // Note: Config doesn't work if contract is instantiated outside of a function or a class
-  const mana = new Mana(MANA_ADDRESS);
-
-  // Note: A real app wouldn't be using a preset private key and hardcoding it!
-  // We're only doing this temporarily while using a hardcoded account with ETH and tokens
-  const ownerPrivKey =
-    "0x11d943d7649fbdeb146dc57bd9cfc80b086bfab2330c7b25651dbaf382392f60";
-  const ownerWallet = createFromPrivateKey(ownerPrivKey);
-  const { address } = beneficiaryWallet;
-  const amountToMint = amountInWei.toString();
-
-  mana.setWallet(ownerWallet);
-  const mintManaAction = mana.mint(address, amountToMint);
-  await mintManaAction.waitForNonceToUpdate();
+export const approveManaSpending = async fromAccount => {
+  const value = 1e18; // one
+  const contracts = getContracts();
+  const { manaContract, marketplaceContract } = contracts;
+  const toAddress = marketplaceContract.getAddress();
+  manaContract.setWallet(fromAccount);
+  const action = manaContract.approve(toAddress, `${value}`);
+  await action.waitForNonceToUpdate();
+};
+
+export const addressesAreEqual = (address1, address2) => {
+  return address1.toUpperCase() === address2.toUpperCase();
+};
+
+export const prepareEstateForSale = async (estateContract, estateForSale) => {
```

Comment:
Is this likely enough to be used on multiple screens that it needs to be extracted to a helpers file?

Reply:
No, not really. I've moved the the helpers to keep the component smaller but if the intent is to keep on helpers only those functions that will be used from many places, `prepareEstateForSale` and `prepareParcelForSale` should back to their component.

Reply:
Yeah, I think functions that are only related to one component still should live in that component. We can always break the component down into more subcomponents as an alternative strategy for keeping components small.

Reply:
The exception to that rule is if a function almost certainly will be used by more components pretty soon, it's fine to extract it even if it's only used by one now

Reply:
Ok, thanks, I'll back they to their component.

Reply:
Great, thanks

Reply:
Done.

Reply:
Thanks! Resolving